projects/observability/src/shared/components/explore-query-editor/order-by/explore-query-order-by-editor.component.ts

```diff
@@ -0,0 +1,243 @@
+            (selectedChange)="this.onAggregationChange($event)"
+            [disabled]="(this.aggregationOptions$ | async)?.length <= 1"
+          >
+            <ht-select-option
+              *ngFor="let aggregationOption of this.aggregationOptions$ | async"
+              [value]="aggregationOption.value"
+              [label]="aggregationOption.label"
+            ></ht-select-option>
+          </ht-select>
+        </div>
+        <div class="order-by-input-container">
+          <span class="order-by-label"> Direction </span>
+          <ht-select
+            [showBorder]="true"
+            [selected]="this.selectedSortBy$ | async"
+            class="order-by-selector"
+            (selectedChange)="this.onSortByChange($event)"
+            [disabled]="(this.selectedMetric$ | async) === undefined"
+          >
+            <ht-select-option
+              *ngFor="let option of this.sortByOptions"
+              [value]="option.value"
+              [label]="option.label"
+            ></ht-select-option>
+          </ht-select>
+        </div>
+      </div>
+    </div>
+  `
+})
+export class ExploreQueryOrderByEditorComponent implements OnChanges {
+  @Input()
+  public orderByExpression?: GraphQlOrderBy;
+
+  @Input()
+  public context?: TraceType;
+
+  @Output()
+  public readonly orderByExpressionChange: EventEmitter<GraphQlOrderBy | undefined> = new EventEmitter();
+
+  public readonly selectedMetric$: Observable<AttributeMetadata | undefined>;
+  public readonly selectedAggregation$: Observable<MetricAggregationType>;
+  public readonly selectedSortBy$: Observable<GraphQlSortDirection>;
+  public readonly metricOptions$: Observable<SelectOption<AttributeMetadata | undefined>[]>;
+  public readonly aggregationOptions$: Observable<SelectOption<MetricAggregationType>[]>;
+  public readonly sortByOptions: SelectOption<GraphQlSortDirection>[] = [
+    {
+      value: SortDirection.Asc,
+      label: 'Asc'
+    },
+    {
+      value: SortDirection.Desc,
+      label: 'Desc'
+    }
+  ];
+  private readonly defaultDirection: GraphQlSortDirection = SortDirection.Asc;
+  private readonly incomingOrderByExpressionSubject: Subject<GraphQlOrderBy> = new ReplaySubject(1);
+  private readonly contextSubject: Subject<TraceType> = new ReplaySubject(1);
+
+  public constructor(private readonly metadataService: MetadataService) {
+    this.selectedAggregation$ = combineLatest([this.contextSubject, this.incomingOrderByExpressionSubject]).pipe(
+      switchMap(() => this.getCurrentSelectedAggregation())
+    );
+
+    this.selectedMetric$ = combineLatest([this.contextSubject, this.incomingOrderByExpressionSubject]).pipe(
+      switchMap(() => this.getCurrentSelectedAttribute())
+    );
+
+    this.selectedSortBy$ = this.incomingOrderByExpressionSubject.pipe(switchMap(() => this.getCurrentSelectedSortBy()));
+
+    this.metricOptions$ = this.contextSubject.pipe(switchMap(context => this.getAttributeOptionsForContext(context)));
```

Comment:
Since i am seeing multiple subscriptions for `metricOptions`, it might be better to add a `share` operator here.

Reply:
added.